src/client/pythonEnvironments/base/locatorUtils.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Uri } from 'vscode';
+import { createDeferred } from '../../common/utils/async';
+import { getURIFilter } from '../../common/utils/misc';
+import { PythonEnvInfo } from './info';
+import {
+    IPythonEnvsIterator,
+    PythonEnvUpdatedEvent,
+    PythonLocatorQuery,
+} from './locator';
+
+/**
+ * Create a filter function to match the given query.
+ */
+export function getQueryFilter(query: PythonLocatorQuery): (env: PythonEnvInfo) => boolean {
+    const kinds = (query.kinds !== undefined && query.kinds.length > 0)
+        ? query.kinds
+        : undefined;
+    let includeNonRooted = true;
+    if (query.searchLocations !== undefined) {
+        if (query.searchLocations.includeNonRooted !== undefined) {
+            includeNonRooted = query.searchLocations.includeNonRooted;
+        } else {
+            // We default to `false`.
+            includeNonRooted = false;
+        }
+    }
+    const locationFilters = getSearchLocationFilters(query);
+    function checkKind(env: PythonEnvInfo): boolean {
+        if (kinds === undefined) {
+            return true;
+        }
+        return kinds.includes(env.kind);
+    }
+    function checkSearchLocation(env: PythonEnvInfo): boolean {
+        if (env.searchLocation === undefined) {
+            // It is not a "rooted" env.
+            return includeNonRooted;
+        } else {
+            // It is a "rooted" env.
+            if (locationFilters === undefined) {
+                return query.searchLocations === undefined;
+            }
+            // Check against the requested roots.
+            return locationFilters.some((filter) => filter(env.searchLocation!));
```

Comment:
```suggestion
            return locationFilters.some((filter) => filter(env.searchLocation));
```
You should be able to remove the `!` here.